cmd/runtime_options.go

```diff
@@ -52,13 +53,16 @@ func runtimeOptionFlagSet(includeSysEnv bool) *pflag.FlagSet {
 	flags := pflag.NewFlagSet("", 0)
 	flags.SortFlags = false
 	flags.Bool("include-system-env-vars", includeSysEnv, "pass the real system environment variables to the runtime")
+	flags.String("compatibility-mode", "extended",
+		`JavaScript compiler compatibility mode, "extended" (ES6+) or "base" (ES5.1+)`)
```

Comment:
I would prefer if this a ... more elaborate, something along the lines of :
base: pure golang JS VM supporting ES5.1+ 
extented: base+ babel with es2015 rreset + core.js v2 - slower and memory consuming but with greater JS support

Reply:
I would prefer leaving those details for a man page or for the official documentation, especially the memory tradeoffs, which will require further explanation anyway.

Flag help text should be brief IMO. Even the parenthesis bothers me here, but at least it's a small hint of the supported feature set.

@na-- @cuonglm What do you think?

Reply:
IMO ... cli help is either brief or helpful. I prefer helpful, mostly because brief leads to googling what the flag meant ... . The `go` tool help is good example - you don't need to google for the vast majority of things, but you will have to read ;) 

Another good thing about it being in the cli flag help is that it will be up to date for the particular version the user is using, not the "latest" documentation (when they are three versions behind) or some article/tutorial that is from a year ago at the point of the reading

But if we won't explain what base and extended mean, then I am for removing the parenthesis as at least it will be less falseful and people (may) google it.

Reply:
I'm also in favor of a longer (though not excessive) explanation in the CLI help, though I'm unsure how well the `pflag` library will handle multi-line help messages

Reply:
> Another good thing about it being in the cli flag help is that it will be up to date for the particular version the user is using

Yeah, that's a good point.

OK, I'll change this to what @MStoykov suggested.

I wouldn't expect issues with newlines and `pflag`, but let's see... :crossed_fingers:

Reply:
Here's how it looks like after a46b247d:

```
...
      --include-system-env-vars           pass the real system environment variables to the runtime (default true)
      --compatibility-mode string         JavaScript compiler compatibility mode, "extended" or "base"
                                          base: pure Golang JS VM supporting ES5.1+
                                          extended: base + Babel with ES2015 preset + core.js v2 - slower and memory consuming but with greater JS support (default "extended")
  -e, --env VAR=value                     add/override environment variable with VAR=value
...
```

Maybe the `extended` line could be wrapped further, and I would prefer to see the `(default "extended")` text immediately after the first line (though this would involve handling default values manually (as done for `--system-tags`), which seems overkill for this detail).

Let me know.

Reply:
Yeah I think another newline after the `-` (maybe even removing it in that case) and I guess if you add another newline at the end `default` will be moved to the next line ? which is definitely better than it being tugged on the end in multiline help IMO

Reply:
Here's how it looks in fe40c53f:

```
...
      --include-system-env-vars           pass the real system environment variables to the runtime (default true)
      --compatibility-mode string         JavaScript compiler compatibility mode, "extended" or "base"
                                          base: pure Golang JS VM supporting ES5.1+
                                          extended: base + Babel with ES2015 preset + core.js v2,
                                                    slower and memory consuming but with greater JS support
                                           (default "extended")
  -e, --env VAR=value                     add/override environment variable with VAR=value
...
```

A bit wonky still, but readable.

Reply:
seems okay for me ... apart from the extra space before the `(default` but ... well it's fine ;)

Reply:
Yes, removing the extra space would require manual handling of default values, which doesn't seem to be worth it for this.